### ISSUE TITLE
Promised returned from setPosition returns immediately

### ISSUE THREAD

**Issue by LandonSchropp** [shaack/cm-chessboard#37]
I'm trying to use the promise returned from `setPosition` to add before and after markers for my moves. However, the promise I'm using seems to return immediately rather than when the animation finishes.

Thanks!

**Comment by LandonSchropp** [comment]
I've traced this problem back to the `initialization` promise. It looks like this promise is being triggered twice. On the first call, the animation callback works correctly. However, on the second resolution, `setPosition` detects that the FEN hasn't changed (even though the first change is still animating), and so it resolves the returned promise immediately.

This is pretty easy to repo by dropping a `console.log` inside of the initialization promise in `setPosition`.

**Comment by LandonSchropp** [comment]
I've traced this problem back to the `initialization` promise inside of Chessboard.js. I'm not sure why, but this promise seems to be resolving twice. Here's how I modified the code:

<img width="468" alt="Screen Shot 2021-04-29 at 6 37 01 PM" src="https://user-images.githubusercontent.com/361591/116630961-e703e200-a919-11eb-9e16-0b3c4be34cc5.png">

And here's what I see in the console:

<img width="633" alt="Screen Shot 2021-04-29 at 6 37 31 PM" src="https://user-images.githubusercontent.com/361591/116630982-f71bc180-a919-11eb-80b8-5e893f4560da.png">

I'm not sure why this is happening, but the second initialize callback causes the chessboard to resolve the promise immediately since it detects that the FEN hasn't chance since the last call.

**Comment by shaack** [comment]
Thats right, if setPosition is called twice with the same position, it resolves immediately. But I ask myself, what should happen? I think, if the animation is running nothing should happen (because the running promise will resolve) and if the animation was finished and the same position was set again, it should resolve immediate. But I am not sure. What do you think?

**Comment by LandonSchropp** [comment]
The issue here is that `setPosition` is only being called once. The promise is still resolving twice.

**Comment by shaack** [comment]
Are you sure? See on this demo page https://shaack.com/projekte/cm-chessboard/examples/pieces-animation.html in the console you see when promises are resolved, the problem only occurs, when a button is pressed twice (and setPosition is called twice.)

**Comment by LandonSchropp** [comment]
Hmm, it looks like your initialization promise is only resolving once during the whole thing (which makes sense), but in mine, it's definitely resolving twice on every call to `setPosition`. Maybe I'm doing something funky somewhere. I'll see if I can get a minimal example working that showcases the problem or fixes my issue.

**Comment by LandonSchropp** [comment]
I'd post my code as-is, but the problem is my code is integrated into a React component, so it's a little hard to just fire it up. 🙂

**Comment by shaack** [comment]
Yeah React ^^

**Comment by shaack** [comment]
But beside from that my question stays, what should happen, if `setPosition` is called twice with the same position.

**Comment by LandonSchropp** [comment]
If it's called twice with the same position, I personally think the second call should have no effect. I would expect the second call to return the promise from the first call, which would resolve when the in-progress animation is complete. I like the idea of an idempotent API here. But I could be biased by my own use case.

As a little background, my use case revolves around using CMChessboard with MobX in React. MobX rerenders the component when the data changes, and in my case, that calls `setPosition` on the board. Currently, I'm controlling those rerenders pretty tightly (I double-check that the FEN has changed before calling `setPosition`), but it would be nice to not have to worry about an errant call to `setPosition` causing an unintended side-effect.

**Comment by shaack** [comment]
You could debounce the call to setPosition to prevent multiple calls.

**Comment by shaack** [comment]
Okay, this works now with version 3.10.11. Setting the position twice will return now the promise of the previous position.

**Comment by LandonSchropp** [comment]
This fixed my issue! Thanks for tackling this. 🙂
